src/formBuilder/FormBuilder.test.js

```diff
@@ -263,4 +263,110 @@ describe('FormBuilder', () => {
       .map((error) => error.text());
     expect(errors).toEqual([]);
   });
+
+  it('validates additionalProperties as a valid property', () => {
+    const jsonSchema = {
+      $schema: `http://json-schema.org/draft-07/schema#`,
+      properties: {},
+      required: [],
+      additionalProperties: false,
+    };
+
+    const props = {
+      schema: JSON.stringify(jsonSchema),
+      uiSchema: '{}',
+      onChange: jest.fn(() => {}),
+      mods: {},
+      className: 'my-form-builder',
+    };
+
+    const div = document.createElement('div');
+    document.body.appendChild(div);
+    const wrapper = mount(<FormBuilder {...props} />, { attachTo: div });
+    const errors = wrapper
+      .find('.alert-warning')
+      .first()
+      .find('li')
+      .map((error) => error.text());
+    expect(errors).toEqual([]);
+  });
+
+  it('support ui:placeholder without schema definition', () => {
+    const jsonSchema = {
+      $schema: 'http://json-schema.org/draft-07/schema#',
+      type: 'object',
+      properties: {
+        input1: {
+          type: 'string',
+          title: 'First Name',
+          description: 'Please enter your first name',
+        },
+      },
+    };
+
+    const uischema = {
+      input1: {
+        'ui:placeholder': 'wow placeholder change',
+      },
+      'ui:order': ['input1'],
+    };
+
+    const props = {
+      schema: JSON.stringify(jsonSchema),
+      uischema: JSON.stringify(uischema),
+      onChange: jest.fn(() => {}),
+      mods: {},
+      className: 'my-form-builder',
+    };
+    const div = document.createElement('div');
+    document.body.appendChild(div);
+    const wrapper = mount(<FormBuilder {...props} />, { attachTo: div });
+    const placeholder = wrapper
+      .find('.card-container')
+      .first()
+      .find('.card-text')
+      .get(6);
+    expect(placeholder.props.value).toEqual('wow placeholder change');
+  });
+
+  it('support ui:placeholder with schema definition', () => {
```

Comment:
Again, not really descriptive of the test assertions.

I'd change to something like:

```suggestion
  it('initializes the FormBuilder with the ui:placeholder specified in the initial uiSchema in definition fields', () => 
```